src/libraries/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -570,53 +553,55 @@ public async Task SendPingAsyncWithHostAndTimeoutAndBufferAndPingOptions()
                 });
         }
 
-        [PlatformSpecific(TestPlatforms.AnyUnix)]  // On Unix, Non-root pings cannot send arbitrary data in the buffer, and do not receive it back in the PingReply.
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
         [Fact]
         public void SendPingWithHostAndTimeoutAndBufferAndPingOptions_Unix()
         {
             IPAddress[] localIpAddresses = TestSettings.GetLocalIPAddresses();
 
-            byte[] buffer = TestSettings.PayloadAsBytes;
+            if (localIpAddresses.Length == 0)
+            {
+                // no local addresses
+                return;
```

Comment:
What would be the case? If anything,  I would do skip (with ConditionalFact) so we can track it.

Reply:
I wasn't sure, so I put it there as sanity check

Reply:
Maybe throw in the helper function...? That would make it at least visible.